src/linking/linking.cpp

```diff
@@ -1186,9 +1187,9 @@ void linkingt::do_type_dependencies(id_sett &needs_to_be_renamed)
 
   std::stack<irep_idt> queue;
 
-  for(id_sett::const_iterator
-      d_it=needs_to_be_renamed.begin();
-      d_it!=needs_to_be_renamed.end();
+  for(std::unordered_set<irep_idt>::const_iterator d_it =
+        needs_to_be_renamed.begin();
+      d_it != needs_to_be_renamed.end();
```

Comment:
Possibly in a separate commit: `std::stack<irep_idt> queue(needs_to_be_renamed.begin(), needs_to_be_renamed.end());` and ditch the `for` loop altogether.

Reply:
It seems that `std::stack` doesn't have a constructor like that. The solution seems to be to use `std::deque` instead.